src/vfile.ts

```diff
@@ -37,18 +39,66 @@ export function dump(vfile: VFile): string {
 /**
  * Read a file path into a virtual file
  *
- * @param path File system path to read
+ * @param path The content, or file system path to read from. If `--`, read from stdin.
  */
-export async function read(path: string): Promise<VFile> {
-  return toVFile.read(path)
+export async function read(content: string): Promise<VFile> {
+  if (!content) throw new Error('Argument `path` is required')
+
+  if (content === '--') {
+    return load(await getStdin())
+  } else if (isPath(content)) {
+    return toVFile.read(content)
+  } else {
+    return load(content)
+  }
 }
 
 /**
- * Write a virtual file to the file system
+ * Write a virtual file to the file system or standard output.
+ *
+ * Will never write an empty file i.e. will only write if the
+ * `vfile` has `contents`.
  *
  * @param vfile Virtual file to write
+ * @param path The file system path to write to, or `--` to write to stdout.
+ */
+export async function write(vfile: VFile, path: string): Promise<void> {
+  if (!path) throw new Error('Argument `path` is required')
+
+  if (path === '--') {
+    console.log(dump(vfile))
+  } else if (path && vfile.contents) {
+    vfile.path = path
+    await toVFile.write(vfile)
+  }
+}
+
+/**
+ * Is a string a file system path?
+ *
+ * Several functions in this module allow for content to be passed as a
+ * string of raw content (e.g. `This is some Markdown`) or as a file system
+ * path. This function, assesses whether a file is a file system path or not.
+ *
+ * If the string starts with `/`, `./`, `../` (or Windows compatible backslash
+ * equivalents as well as drive letter prefixed strings e.g. `C:\`)
+ * then it is assumed to be a path but the presence of the path is not checked.
+ * The reason for not checking presence
+ * here is because "if the content looks like a path then the user probably meant
+ * it to be a path". That is, if a user tries to convert the string "./file.md" then,
+ * if that file doesn't exist, she probably wants the Markdown converter to give me an error
+ * message. She probably doesn't want the `match` function to try and find some other converter
```

Comment:
I like that we're being gender inclusive in our writing and wanted to bring up a stylistic comment for this and our other documentation.
We could go a step further and use the [singular they](https://blog.ap.org/products-and-services/making-a-case-for-a-singular-they) in our documentation to be inclusive of gender non-binary people.

Reply:
:+1: